pkg/daemon/controller/controller.go

```diff
@@ -252,15 +257,18 @@ func (sc *ServiceController) Reconcile(_ context.Context, req ctrl.Request) (res
 		return ctrl.Result{}, err
 	}
 
-	awaitingRestart, err := sc.reconcileEnvironmentVariables(cmData.EnvironmentVars, node)
+	restartRequired, err := reconcileEnvironmentVariables(cmData.EnvironmentVars)
 	if err != nil {
 		return ctrl.Result{}, err
 	}
-	if awaitingRestart {
-		// if a restart is required, there is nothing more the controller can do until the instance is rebooted
+	if restartRequired {
+		// If there's any changes, an instance restart is required to ensure all processes pick up the updates.
+		// Applying the reboot annotation results in an event picked up by WMCO's node controller to reboot the instance.
+		// If a restart is required, there is nothing more the controller can do until the instance is rebooted
 		klog.Info("waiting for reboot")
```

Comment:
Sure I can add it, the "reboot completed" log will be from WMCO's side. For WICD, we will know reboot is done because 1. service will restart and 2. it will start reacting to node events again